class.jetpack-cli.php

```diff
@@ -918,6 +922,28 @@ public function partner_provision( $args, $named_args ) {
 			$this->partner_provision_error( new WP_Error( 'missing_access_token', __( 'Missing or invalid access token', 'jetpack' ) ) );
 		}
 
+		$url_args = array(
+			'home_url' => 'WP_HOME',
+			'site_url' => 'WP_SITEURL',
+		);
+
+		foreach ( $url_args as $url_arg => $constant_name ) {
+			// Anonymous functions were introduced in 5.3.0. So, if we're running on
+			// >= 5.3.0, use an anonymous function to set the home/siteurl values.
+			//
+			// Otherwise, fallback to setting the home/siteurl value via the WP_HOME and
+			// WP_SITEURL constants if the constant hasn't already been defined.
+			if ( isset( $named_args[ $url_arg ] ) ) {
```

Comment:
In the case we're talking about in this ticket the partner is setting via constants. So why introduce the option to filter here, plus the complexity of checking the PHP version? Maybe for now we can just define the constants.

Reply:
The issue I see with relying solely on constants, is that if the host has set the constant already, then we can't set it again. Unfortunately, I'm not 100% sure how the host is setting the constant. In this case, I could ask. But, that may not scale as we add more hosting partners.

To be sure that we can set the home and siteurl values as often as possible, we'll need to rely on the filter.